controllers/provider-openstack/pkg/controller/controlplane/valuesprovider_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+	"encoding/json"
+	openstackv1aplha "github.com/gardener/gardener-extensions/controllers/provider-openstack/pkg/apis/openstack/v1alpha1"
+	"github.com/gardener/gardener-extensions/controllers/provider-openstack/pkg/openstack"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	mockclient "github.com/gardener/gardener-extensions/pkg/mock/controller-runtime/client"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/operation/common"
+
+	"github.com/golang/mock/gomock"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+const (
+	namespace = "test"
+)
+
+var (
+	dhcpDomain     = "dhcp-domain"
+	requestTimeout = "2s"
+)
+
+var _ = Describe("ValuesProvider", func() {
+	var (
+		ctrl *gomock.Controller
+
+		// Build scheme
+		scheme = runtime.NewScheme()
+		_      = openstackv1aplha.AddToScheme(scheme)
+
+		cp = &extensionsv1alpha1.ControlPlane{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "control-plane",
+				Namespace: namespace,
+			},
+			Spec: extensionsv1alpha1.ControlPlaneSpec{
+				SecretRef: corev1.SecretReference{
+					Name:      common.CloudProviderSecretName,
+					Namespace: namespace,
+				},
+				ProviderConfig: &runtime.RawExtension{
+					Raw: encode(&openstackv1aplha.ControlPlaneConfig{
+						LoadBalancerProvider: "load-balancer-provider",
+						DHCPDomain:           &dhcpDomain,
```

Comment:
You could use `util.StringPtr` to convert a string literal, e.g. "dhcp-domain" into a pointer directly.